src/main/java/io/github/bonigarcia/wdm/WebDriverManager.java

```diff
@@ -1013,25 +1013,23 @@ protected Integer versionCompare(String str1, String str2) {
         }
 
         String driverStr = driverUrl.toString();
-        String driverUrlContent = driverUrl.getPath();
 
         HttpResponse response = httpClient
                 .execute(httpClient.createHttpGet(driverUrl));
         try (InputStream in = response.getEntity().getContent()) {
-            org.jsoup.nodes.Document doc = Jsoup.parse(in, null, "");
+            org.jsoup.nodes.Document doc = Jsoup.parse(in, null, driverStr);
             Iterator<org.jsoup.nodes.Element> iterator = doc.select("a")
                     .iterator();
             List<URL> urlList = new ArrayList<>();
 
             while (iterator.hasNext()) {
-                String link = iterator.next().attr("href");
-                if (link.contains("mirror") && link.endsWith(SLASH)) {
-                    urlList.addAll(getDriversFromMirror(new URL(
-                            driverStr + link.replace(driverUrlContent, ""))));
-                } else if (link.startsWith(driverUrlContent)
-                        && !link.contains("icons")) {
-                    urlList.add(new URL(
-                            driverStr + link.replace(driverUrlContent, "")));
+                String link = iterator.next().attr("abs:href");
+                if (link.startsWith(driverStr) && link.endsWith(SLASH)) {
+                    urlList.addAll(getDriversFromMirror(new URL(link)));
+                } else if (link.startsWith(driverStr) && !link.contains("icons") && (link.toLowerCase().endsWith(".bz2")
```

Comment:
Is this condition actually needed? I mean the check about ending with bz2, zip, etc. If so, the following extension should also be added: msi (it is for backwards compatibility with old versions of edgedriver).